From 4a4796f844645be3f4c43d9d301bdd260669f8e9 Mon Sep 17 00:00:00 2001
From: Callum <38501234+calumari@users.noreply.github.com>
Date: Mon, 25 May 2020 08:36:45 +0100
Subject: [PATCH] EntityLookAtPlayerEvent

---
 .../java/net/minecraft/server/PathfinderGoalLookAtPlayer.java    | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/main/java/net/minecraft/server/PathfinderGoalLookAtPlayer.java b/src/main/java/net/minecraft/server/PathfinderGoalLookAtPlayer.java
index a73e5b77ca..f2cc5cff83 100644
--- a/src/main/java/net/minecraft/server/PathfinderGoalLookAtPlayer.java
+++ b/src/main/java/net/minecraft/server/PathfinderGoalLookAtPlayer.java
@@ -53,6 +53,7 @@ public class PathfinderGoalLookAtPlayer extends PathfinderGoal {
 
     @Override
     public boolean b() {
+        if (this.b instanceof EntityPlayer && !new net.socialhangover.conduit.event.EntityLookAtPlayerEvent(this.a.getBukkitLivingEntity(), ((EntityPlayer) this.b).getBukkitEntity()).callEvent()) { return false; } // Conduit
         return !this.b.isAlive() ? false : (this.a.h(this.b) > (double) (this.c * this.c) ? false : this.g > 0);
     }
 
-- 
2.24.1.windows.2

