From 0129c6655a19a03df676f68596671bb158cbbeda Mon Sep 17 00:00:00 2001
From: Callum <38501234+calumari@users.noreply.github.com>
Date: Mon, 25 May 2020 08:36:45 +0100
Subject: [PATCH] Add EntityLookAtPlayerEvent

---
 .../net/minecraft/server/PathfinderGoalLookAtPlayer.java  | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/main/java/net/minecraft/server/PathfinderGoalLookAtPlayer.java b/src/main/java/net/minecraft/server/PathfinderGoalLookAtPlayer.java
index a73e5b77c..e6396ecd8 100644
--- a/src/main/java/net/minecraft/server/PathfinderGoalLookAtPlayer.java
+++ b/src/main/java/net/minecraft/server/PathfinderGoalLookAtPlayer.java
@@ -53,6 +53,14 @@ public class PathfinderGoalLookAtPlayer extends PathfinderGoal {
 
     @Override
     public boolean b() {
+        // Conduit start
+        if (this.b instanceof EntityPlayer) {
+            net.socialhangover.conduit.events.EntityLookAtPlayerEvent event = new net.socialhangover.conduit.events.EntityLookAtPlayerEvent(this.a.getBukkitLivingEntity(), ((EntityPlayer) this.b).getBukkitEntity());
+            if (event.isCancelled()) {
+                return false;
+            }
+        }
+        // Conduit end
         return !this.b.isAlive() ? false : (this.a.h(this.b) > (double) (this.c * this.c) ? false : this.g > 0);
     }
 
-- 
2.24.1.windows.2

