From 4e58230f326eb090d2b04498e7580baa0aa3565a Mon Sep 17 00:00:00 2001
From: Callum <38501234+calumari@users.noreply.github.com>
Date: Thu, 9 Jul 2020 15:59:50 +0100
Subject: [PATCH] PlayerProfile getProperty

---
 .../destroystokyo/paper/profile/CraftPlayerProfile.java   | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/main/java/com/destroystokyo/paper/profile/CraftPlayerProfile.java b/src/main/java/com/destroystokyo/paper/profile/CraftPlayerProfile.java
index 2751ce7f15..2001f45d46 100644
--- a/src/main/java/com/destroystokyo/paper/profile/CraftPlayerProfile.java
+++ b/src/main/java/com/destroystokyo/paper/profile/CraftPlayerProfile.java
@@ -43,6 +43,14 @@ public class CraftPlayerProfile implements PlayerProfile {
         return profile.getProperties().containsKey(property);
     }
 
+    // Conduit start
+    @Override
+    public ProfileProperty getProperty(String property) {
+        Property value = com.google.common.collect.Iterables.getFirst(profile.getProperties().get(property), null);
+        return value == null ? null : new ProfileProperty(value.getName(), value.getValue(), value.getSignature());
+    }
+    // Conduit end
+
     @Override
     public void setProperty(ProfileProperty property) {
         String name = property.getName();
-- 
2.28.0.windows.1

