From 14000136486c3dc481689a24771d4ac82f81d7f4 Mon Sep 17 00:00:00 2001
From: Callum <38501234+calumari@users.noreply.github.com>
Date: Wed, 26 Aug 2020 21:52:47 +0100
Subject: [PATCH] Expose HumanEntity dropItem

---
 .../org/bukkit/craftbukkit/entity/CraftHumanEntity.java   | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftHumanEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftHumanEntity.java
index 53917d3b38..7a37050755 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftHumanEntity.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftHumanEntity.java
@@ -697,4 +697,12 @@ public class CraftHumanEntity extends CraftLivingEntity implements HumanEntity {
     public boolean dropItem(boolean dropAll) {
         return getHandle().dropItem(dropAll);
     }
+
+    // Paper start
+    @Override
+    public org.bukkit.entity.Item dropItem(ItemStack itemStack) {
+        net.minecraft.server.EntityItem entity = getHandle().drop(CraftItemStack.asNMSCopy(itemStack), true);
+        return entity == null ? null : (org.bukkit.entity.Item) entity.getBukkitEntity();
+    }
+    // Paper end
 }
-- 
2.28.0.windows.1

